### PR TITLE
[BDTT-21] Customer viewing vessel location

### PR DESCRIPTION
**Pull request by prussianman**
# Description

Fixes [BDTT-21](https://bottlenose.atlassian.net/browse/BDTT-21)

image

List all acceptance criteria of user story and indicate which has been solved previously and which is solved currently

- [x] Vessel Location Microservice returns current coords and destination coords
- [x] Complex Microservice gracefully handles failure to return coords
- [x] Vessel location is displayed
- [x] Destination port is displayed
- [ ] Route between them is displayed 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor
- [ ] Migration

# Checklist:

- [x] I have added the appropriate labels to my PR
- [x] My code follows the style guidelines of this project
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: services/frontend/app/src/components/ViewShipments/ViewShipmentComponent.jsx [hunk 1]
```diff
@@ -142,12 +143,13 @@ function ShipmentCard({ item, index, setLoading }) {
             containerReleaseDateTime: result.cont_released,
             deliveryTakenDateTime: result.del_taken,
             shippingLine: result.shipping_line,
-            direction: directionType
-          }
-        })
+            direction: item.type,
```
Comment (charmjj) [inline services/frontend/app/src/components/ViewShipments/ViewShipmentComponent.jsx:146]: Stick to "direction: directionType"

File: services/frontend/app/src/components/BLStatus/VesselView.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      >
+        <MarkerF position={{ lat: originLat, lng: originLong }} 
+        icon={shippingIcon}
+      /> 
+        <MarkerF position={{ lat: destLat, lng: destLong }}
+        />
+        <InfoWindowF
+      onLoad={onLoad}
+      position={{lat: destLat, lng: destLong }}>
+      <div style={divStyle}>
```
Comment (charmjj) [inline services/frontend/app/src/components/BLStatus/VesselView.jsx:53]: Optional: can just use tailwind classes

className="p-8 bg-white"

File: services/frontend/app/src/components/BLStatus/VesselView.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  console.log("infoBox: ", infoBox)
+};
+
+const divStyle = {
+  background: "white",
+  // border: "1px solid #ccc",
+  padding: 8
+}
+
+// "https://cdn-icons-png.flaticon.com/128/1656/1656475.png"
```
Comment (charmjj) [inline services/frontend/app/src/components/BLStatus/VesselView.jsx:21]: Remove unnecessary comments

**Review by charmjj — CHANGES_REQUESTED** [review]

**Review by pandaponn — APPROVED** [review]
Feature works as intended, albeit with hard-coded coordinates. Unfortunately, due to a lack of fresh BL numbers, the APIs are not able to retrieve destination coordinates which breaks the front end and there is not much we can do about this. Do consider adding error handling for such scenarios. 

Nonetheless, great work in such a short delivery period.

File: services/backend/core/vessel_location.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+        try: 
+            # sea routes api, low rate limit
+            cords, destination_cords = get_vessel_location(imo, VESSEL_API_KEY)
+            print("*** Using API provider: SEA ROUTES")
+
+        except:
+            print("*** Using API provider: ZYLA")
+            #if sea routes api fails, use zyla api, seems to be okay but takes way longer and does not return destination coords
+            cords = get_vessel_location2(imo, LOCATION_API_KEY)
+            destination_cords = []
```
Comment (pandaponn) [inline services/backend/core/vessel_location.py:90]: Consider adding error handling when destination_coords are unretrievable as it breaks the front end

**Pull request merged**